external/wakaama/examples/client/lwm2mclient.c

```diff
@@ -75,1474 +80,804 @@
-        else
-        {
-            fprintf(stderr, "write not supported for specified resource!\n");
-        }
-        return;
-    }
-    else
-    {
-        fprintf(stderr, "Object not found !\n");
-    }
+	g_quit = 2;
 }
 
-static coap_protocol_t coap_get_protocol_from_uri(const char *uri)
+void handle_value_changed(lwm2m_context_t *lwm2mH,
+						  lwm2m_uri_t *uri,
+						  const char *value,
+						  size_t valueLength)
 {
-  coap_protocol_t type;
-  for (type = COAP_UDP; type < COAP_PROTOCOL_MAX; type++) {
-    if (!strncmp(uri, coap_uri_prefix[type], strlen(coap_uri_prefix[type]))) {
-		return type;
+	lwm2m_object_t *object = (lwm2m_object_t *)LWM2M_LIST_FIND(lwm2mH->objectList, uri->objectId);
+
+	if (NULL != object) {
+		if (object->writeFunc != NULL) {
+			lwm2m_data_t *dataP;
+			int result;
+
+			dataP = lwm2m_data_new(1);
+			if (dataP == NULL) {
+				fprintf(stderr, "Internal allocation failure !\n");
+				return;
+			}
+			dataP->id = uri->resourceId;
+			lwm2m_data_encode_nstring(value, valueLength, dataP);
+
+			result = object->writeFunc(uri->instanceId, 1, dataP, object);
+			if (COAP_405_METHOD_NOT_ALLOWED == result) {
+				switch (uri->objectId) {
+				case LWM2M_DEVICE_OBJECT_ID:
+					result = device_change(dataP, object);
+					break;
+				default:
+					break;
+				}
+			}
+
+			if (COAP_204_CHANGED != result) {
+				fprintf(stderr, "Failed to change value!\n");
+			} else {
+				fprintf(stderr, "value changed!\n");
+				lwm2m_resource_value_changed(lwm2mH, uri);
+			}
+			lwm2m_data_free(1, dataP);
+			return;
+		} else {
+			fprintf(stderr, "write not supported for specified resource!\n");
+		}
+		return;
+	} else {
+		fprintf(stderr, "Object not found !\n");
 	}
-  }
-  return type;
 }
 
 #ifdef WITH_TINYDTLS
-void * lwm2m_connect_server(uint16_t secObjInstID,
-                            void * userData)
+void *lwm2m_connect_server(uint16_t secObjInstID,
+						   void *userData)
 {
-  client_data_t * dataP;
-  lwm2m_list_t * instance;
-  dtls_connection_t * newConnP = NULL;
-  dataP = (client_data_t *)userData;
-  lwm2m_object_t  * securityObj = dataP->securityObjP;
-
-  instance = LWM2M_LIST_FIND(dataP->securityObjP->instanceList, secObjInstID);
-  if (instance == NULL) return NULL;
-
-  newConnP = connection_create(dataP->connList, dataP->sock, securityObj, instance->id, dataP->lwm2mH, dataP->addressFamily);
-  if (newConnP == NULL)
-  {
-      fprintf(stderr, "Connection creation failed.\n");
-      return NULL;
-  }
-
-  dataP->connList = newConnP;
-  return (void *)newConnP;
+	client_data_t *dataP;
+	lwm2m_list_t *instance;
+	dtls_connection_t *newConnP = NULL;
+	dataP = (client_data_t *)userData;
+	lwm2m_object_t   *securityObj = dataP->securityObjP;
+
+	instance = LWM2M_LIST_FIND(dataP->securityObjP->instanceList, secObjInstID);
+	if (instance == NULL) {
+		return NULL;
+	}
+
+
+	newConnP = connection_create(dataP->connList, dataP->sock, securityObj, instance->id, dataP->lwm2mH, dataP->addressFamily);
```

Comment:
Below code is a latest one. 

fprintf(stderr, "Opening connection to server at %s:%s\r\n", host, port);
newConnP = connection_create(COAP_UDP, dataP->connList, dataP->sock, host, port, dataP->addressFamily);